astra/src/main/java/com/slack/astra/chunkManager/CachingChunkManager.java

```diff
@@ -126,4 +180,76 @@ public void addMessage(Trace.Span message, long msgSize, String kafkaPartitionId
     throw new UnsupportedOperationException(
         "Adding messages is not supported on a caching chunk manager");
   }
+
+  private void onAssignmentHandler(CacheNodeAssignment assignment) {
+    if (Objects.equals(assignment.cacheNodeId, this.cacheNodeId)) {
+      LOG.info(
+          "Assignment handler fired for cache node {} and assignment {}",
+          cacheNodeId,
+          assignment.assignmentId);
+      Map<String, SnapshotMetadata> snapshotsBySnapshotId =
+          snapshotMetadataBySnapshotId(snapshotMetadataStore);
+      try {
+        if (chunksMap.containsKey(assignment.assignmentId)) {
+          ReadOnlyChunkImpl<T> chunk = chunksMap.get(assignment.assignmentId);
+
+          if (chunkStateChangedToEvict(assignment, chunk)) {
+            LOG.info(
+                "Starting eviction for assignment {} from node {}",
+                assignment.assignmentId,
+                cacheNodeId);
+            Thread.ofVirtual().start(() -> chunk.evictChunk(assignment));
```

Comment:
Doing this async may end up causing some issues - because you've only send the evictChunk to be executed, but it hasn't yet been completed. Note the `onAssignmentHandler` already runs async as part of the zookeeper event handler.